tests/integration/cartography/intel/github/test_repos.py

```diff
@@ -8,423 +10,398 @@
 TEST_UPDATE_TAG = 123456789
 TEST_JOB_PARAMS = {"UPDATE_TAG": TEST_UPDATE_TAG}
 TEST_GITHUB_URL = "https://fake.github.net/graphql/"
-
-
-def _ensure_local_neo4j_has_test_data(neo4j_session):
-    repo_data = cartography.intel.github.repos.transform(
-        GET_REPOS,
-        DIRECT_COLLABORATORS,
-        OUTSIDE_COLLABORATORS,
-    )
-    cartography.intel.github.repos.load(
+FAKE_API_KEY = "fake-key"
+TEST_ORG = "simpsoncorp"
+
+
+@patch.object(
+    cartography.intel.github.repos,
+    "get",
+    return_value=GET_REPOS,
+)
+@patch.object(
+    cartography.intel.github.repos,
+    "_get_repo_collaborators_for_multiple_repos",
+    side_effect=[DIRECT_COLLABORATORS, OUTSIDE_COLLABORATORS],
+)
+def test_sync_repos(mock_collabs, mock_get, neo4j_session):
+    # Act - setup data using the sync function
+    cartography.intel.github.repos.sync(
         neo4j_session,
         TEST_JOB_PARAMS,
-        repo_data,
+        FAKE_API_KEY,
+        TEST_GITHUB_URL,
+        TEST_ORG,
     )
 
-
-def test_transform_and_load_repositories(neo4j_session):
+    # Assert if all repository nodes and relationships are created as expected
+    _test_repository_nodes(neo4j_session)  # GitHubRepository nodes
+    _test_owner_nodes_and_relationships(
+        neo4j_session
+    )  # GitHubRepository -[:OWNER]-> GitHubOrganization/GitHubUser
+    _test_branch_relationships(
+        neo4j_session
+    )  # GitHubRepository -[:BRANCH]-> GitHubBranch
+    _test_language_relationships(
+        neo4j_session
+    )  # GitHubRepository -[:LANGUAGE]-> ProgrammingLanguage
+    _test_collaborator_relationships(
+        neo4j_session
+    )  # GitHubUser -[:DIRECT_COLLAB_*, OUTSIDE_COLLAB_*]-> GitHubRepository
+    _test_python_dependencies(
+        neo4j_session
+    )  # GitHubRepository -[:REQUIRES]-> PythonLibrary
+    _test_github_dependencies(
+        neo4j_session
+    )  # GitHubRepository -[:REQUIRES]-> Dependency
+    _test_dependency_manifests(
+        neo4j_session
+    )  # GitHubRepository -[:HAS_MANIFEST]-> DependencyGraphManifest -[:HAS_DEP]-> Dependency
+    _test_dependency_relationship_properties(
+        neo4j_session
+    )  # REQUIRES relationship properties
```

Comment:
Please follow https://github.com/cartography-cncf/cartography/blob/fd1fb47b06a1bc2f23382998814ff7de49cf64d1/tests/integration/cartography/intel/lastpass/test_users.py#L17 as an example. No need to introduce `_test()` functions because when a test fails, we need to go deeper by one level. It is ok for the test function to be long.